### ISSUE TITLE
Test android app upgrade for user who have `> 1 MYSTT`

### ISSUE THREAD

**Issue by chompomonim** [mysteriumnetwork/node#3779]
This scenario is for users who have topuped their dVPN app balance using our CoinGate integration (on testnet2). 

## Scenario to test
1. Topup your dVPN app balance using BTC, LTC, ETH, MYST or other supported cryptos (could be done a while ago, your current balance should be more than 1 MYSTT).

1. Download your private key. You can do that in `menu -> Profile` screen (you'll need it later during testnet3 app installation).

1. Download newest testnet3 app (.apk file) from [https://github.com/mysteriumnetwork/nightly/releases/latest](https://github.com/mysteriumnetwork/nightly/releases/latest) and install it. (you may need to uninstall existing Mysterium dVPN app via google play store)

1. Launch the app. During onboarding step import your private keys (don't create new one).

1. After finishing onboarding go to menu → account and check your app balance, it should remain the same as before upgrade.

---------- 
**NOTE**: After finishing this scenario please post a comment above (if all was ok, please also give us know about that).

**Comment by vkuznecovas** [comment]
https://testnet3-hermes.mysterium.network/api/v1/data/consumer/0xc105bce8d93848441ea5d8cae9bdf0b47a1005ae
^ migrated from testnet2 with 2.5 for balance.

**Comment by vkuznecovas** [comment]
Thing to note though, the balance did not update until I connected to a node.

**Comment by creotiv** [comment]
Found issue on android 9 (kernel 4.9.117)
Cant select keystore.json (or just keystore) to import account. Trying to figure out the root of the problem

**Comment by creotiv** [comment]
Pushed fix for android https://github.com/mysteriumnetwork/mysterium-vpn-mobile/pull/441
Migrated without any problems with balance 4.0: https://testnet3-hermes.mysterium.network/api/v1/data/consumer/0xfe0f1b0bff205beb4c15d348237311219728899c

**Comment by GoZippy** [comment]
During topup - issue with only method to topup using imbedded coinbank link api generated. For anonymous topup is impossible with third party. Also - no fee for network shown so no way to know cost to use third party not affiliated with.

I tried BTC, BTC lightning, ETH all using same process flow - so gave up.

Note: If using mobile app to topup, the form only presents user with the QR or a copy to clipboard for the coinbank invoice link with timer. Q: If user only has one phone - how are they going to scan it with their wallet address? I had to install MetaMask on mobile (Android - Samsung s9+ test unit) and sync wallets to then attempt again within browser by opening the presented coinbase link and then manually entering ETH to send. This is WAY too long a process to go mainstream if you ask me - then there is no confirmation or info on network fee. I trued to use a wallet for ETH with $20 in it or there about and it said insuficient funds even for 20 mystt (I had .96 or so on wallet in app already).

I also tried with BTC but again no option to check fees. Not worth $10 or more transaction fee to add $7 of mystt. Tried again with lightning network enabled and again locked into the same process flow. 

It is nice to see simple api use to return info back but I feel it is injecting dependency on third party (even trusted) that need not be there. Any way to integrate swap in-app with link to osmosis like or gravity or allow direct lightning contract connections with mobile app and linked lightning... anyhow - I did not complete top-up as result. Will grab other phone and send BTC from other account but not excited about spending that much in transaction fee and waiting around forever for confirmation and still not knowing how coinbank is storing or using our data - terms of service no shown and no info on how they use data or track or otherwise... very concerned with abstracting away payment gateway when we have an amazing decentralized privacy focused app to begin with.

So, sadly, this said I could not topup to over 1MYST so I quit the test there. If my spouse wants to loan me some ETH from her account on her phone I will try again later.

**Comment by chompomonim** [comment]
@GoZippy I passed your suggestions to dVPN app team. However I'm afraid that todo it in your suggested way will include quite a big of effort from their side. Also I don't think that there is privacy issues with our current implementation, also I don't think that in app swap integration (e.g. uniswap or quickwap, osmosis is Cosmos stuff, for Lightning Network you would still need some 3rd party to convert from BTC to MYST) would add any more privacy. It would maybe avoid dependancy on 3rd party provider and would give less chance for someone to censor/stop your topup. 

I just created QuickSwap integration proposal for topup flow.
https://github.com/mysteriumnetwork/mysterium-vpn-desktop/issues/182
